### PR TITLE
Tickets/dm 34086

### PR DESCRIPTION
**Pull request by pcortes80**

### CODE REVIEW

File: doc/userManual.md [hunk 1]
```diff
@@ -184,8 +184,10 @@ Please define the telemetry with the `Telemetry Id` and `telDetails` strings, an
 ## Connection Status
 
 There is a VI called **getClientStatus&#46;vi** that allows you to know whether the connections between the TCP clients and TCP server are alive or not.
-If all the TCP clients connect to the TCP servers, a `GenRead` user event is triggered with the string message `Clients are on`.
-If one of the TCP clients disconnects, a `GenRead` user event is triggered with the string message `Clients are off`.
+It uses the **TcpServerBase.connStatusCluster.ctl** with the `areClientsConnected` enum, and the `connStatusMsg` string. 
+The enum has three options: (1) unchanged, (2) connected, and (3) unconnected. 
+If all the TCP clients connect to the TCP servers, a `GenRead` user event is triggered with the string message `Clients are on` and the `connected` value in the enum.
+If one of the TCP clients disconnects, a `GenRead` user event is triggered with the string message `Clients are off` and the `unconnected` value in the enum.
```
Comment (teweitsai) [inline doc/userManual.md:191]: Please add the following sentence if you do not have it:

```
The **TcpServer** will report the internal error or message by `connStatusMsg` string, and the enum of `areClientsConnected` will be "unchanged" at that time.
If not, there should be something wrong in the **TcpServer**.
```

File: doc/versionHistory.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Version History
 
+0.3.9
+
+- Replace `areClientsConnected` boolean by `areClientsConnected` enum in **TcpServerBase.connStatusCluster.ctl**. 
+- Update `README.md` in the `src/` directory.
+- Update `userManual.md` in the `doc/` directory.
+
```
Comment (teweitsai) [inline doc/versionHistory.md:8]: Please add the following:

```
The user needs to update the use of **GenRead** user event to judge the connection is on or off with the `areClientsConnected` enum.
```

File: src/README.md [hunk 3]
```diff
@@ -124,6 +124,8 @@ If either command or telemetry instance of Python TCP Client disconnects from TC
 If all Python TCP Client instances are connected again, a generic user event is sent.
 - The `TcpServerBase.sendErrorExplanation.vi` is used to send a specific and important error to the ComponentSimulator.
 This uses the `GenRead` user event.
+- The `TcpServer.getClientStatus.vi` gets the notification from TCP Server that TCP Client connects or disconnects and then sends an user event to the Component.
+This uses the `GenRead` user event.
```
Comment (teweitsai) [inline src/README.md:128]: This uses the `GenRead` user event with the enum of `areClientsConnected`.

**Review by teweitsai — APPROVED** [review]
Looks good. Please help to improve the documents based on the comments. Thanks!

File: doc/userManual.md (hunk 1 above)
Comment (pcortes80) [reply doc/userManual.md:191]: Done.

File: src/README.md (hunk 3 above)
Comment (pcortes80) [reply src/README.md:128]: Done.

File: doc/versionHistory.md (hunk 2 above)
Comment (pcortes80) [reply doc/versionHistory.md:8]: Done.

**Pull request merged**